From 65faafd2b52030e447be5da63eb0116d9c098141 Mon Sep 17 00:00:00 2001
From: huzhen <george.hu@spacemit.com>
Date: Sun, 28 Apr 2024 21:36:30 +0800
Subject: [PATCH 0619/1218] nvme: change the segment size of io request queue
 for nvme on k1

  Because the swiotlb buffer size is 512k, for nvme ssd devices,
  the maximum length of a single dma sg in the io request queue can only be 512k.

Change-Id: I7b02572742dd5b9ff26de6e469dce71c20741aae
---
 drivers/nvme/host/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index 895fb163d48e..d24c55e90239 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -2380,6 +2380,10 @@ static int nvme_update_ns_info_block(struct nvme_ns *ns,
 	if (!nvme_update_disk_info(ns, id, &lim))
 		capacity = 0;
 
+#if defined(CONFIG_SOC_SPACEMIT_K1X)
+	blk_queue_max_segment_size(ns->disk->queue, 0x80000);
+#endif
+
 	nvme_config_discard(ns, &lim);
 	if (IS_ENABLED(CONFIG_BLK_DEV_ZONED) &&
 	    ns->head->ids.csi == NVME_CSI_ZNS)
-- 
2.50.1

